sqlmesh/core/plan/builder.py

```diff
@@ -578,21 +578,25 @@ def _categorize_snapshot(
                     if mode == AutoCategorizationMode.FULL:
                         snapshot.categorize_as(SnapshotChangeCategory.BREAKING)
         elif self._context_diff.indirectly_modified(snapshot.name):
-            categories = []
+            all_upstream_categories = []
+            direct_parent_categories = []
 
             for p_id in dag.upstream(s_id):
                 parent = self._context_diff.snapshots.get(p_id)
 
                 if parent and self._is_new_snapshot(parent):
-                    categories.append(parent.change_category)
+                    all_upstream_categories.append(parent.change_category)
+                    if p_id in snapshot.parents:
+                        direct_parent_categories.append(parent.change_category)
 
-            if not categories or any(
+            if not direct_parent_categories or any(
```

Comment:
if there are no direct parent categories, but then grandparent A is forward only, is this still correct?

Reply:
How can there be no direct parent categories?

Reply:
Note: it's direct **parent** not direct **category**

Reply:
Sorry, let me rephrase the question. How can there be a grandparent category but not a parent category?

Reply:
i’m not sure, is there a case of a revert or something where it’s possible?